rootex/core/resource_loader.cpp

```diff
@@ -91,6 +91,102 @@ int ResourceLoader::Preload(ResourceCollection paths, Atomic<int>& progress)
 	Application::GetSingleton()->getThreadPool().submit(tasks);
 
 	return tasks.size() - 1; // One less for the dummy task
+}
+
+int ResourceLoader::Export(const String& sceneName, const String& sceneFilePath, Atomic<int>& progress)
```

Comment:
This function should set `progress` to 0 as soon as the control enters here because in case the exporting failed, the function returns 0 so the progress should be 0 as well